lib/provable-generic.ts

```diff
@@ -178,6 +178,30 @@ function createDerivers<Field>(): {
 
       if (isProvable(typeObj)) return typeObj.check(obj);
 
+      if (typeof typeObj === 'function') {
+        throw new Error(
+          `provable: invalid type detected. Functions are not supported as types. ` +
+            `Ensure you are passing an instance of a supported type or an anonymous object.\n`
+        );
+      }
```

Comment:
Struct is a function though, so this has to come after the detailed error about Struct!

Reply:
[Resolved!](https://github.com/o1-labs/o1js-bindings/pull/278/commits/cbdbff3d9c317aa1131667cd56f2a8a374640436)